Test/Utils/ContrailAPI/ConfigureDNS.ps1

```diff
@@ -0,0 +1,174 @@
+    Param (
+        [Parameter(Mandatory = $true)] [string] $ContrailUrl,
+        [Parameter(Mandatory = $true)] [string] $AuthToken,
+        [Parameter(Mandatory = $true)] [string] $DNSRecordUuid
+    )
+
+    $DNSRecordURl = $ContrailUrl + "/virtual-DNS-record/" + $DNSRecordUuid
+    Invoke-RestMethod -Method Delete -Uri $DNSRecordURl -Headers @{"X-Auth-Token" = $AuthToken} | Out-Null
+}
+
+function Add-ContrailDNSServer {
+    Param (
+        [Parameter(Mandatory = $true)] [string] $ContrailUrl,
+        [Parameter(Mandatory = $true)] [string] $AuthToken,
+        [Parameter(Mandatory = $true)] [string] $DNSServerName,
+        [Parameter(Mandatory = $false)] [VirtualDNSData] $VirtualDNSData = [VirtualDNSData]::new()
+    )
+
+    $VirtualDNS = @{
+        domain_name                 = $VirtualDNSData.DomainName
+        dynamic_records_from_client = $VirtualDNSData.DynamicRecordsFromClient
+        record_order                = $VirtualDNSData.RecordOrder
+        default_ttl_seconds         = $VirtualDNSData.DefaultTTL
+        floating_ip_record          = $VirtualDNSData.FloatingIpRecord
+        external_visible            = $VirtualDNSData.ExternalVisible
+        reverse_resolution          = $VirtualDNSData.ReverseResolution
+    }
+
+    $Request = @{
+        "virtual-DNS" = @{
+            parent_type      = "domain"
+            fq_name          = @("default-domain", $DNSServerName)
+            virtual_DNS_data = $VirtualDNS
+        }
+    }
+
+    $RequestUrl = $ContrailUrl + "/virtual-DNSs"
+    $Response = Invoke-RestMethod -Uri $RequestUrl -Headers @{"X-Auth-Token" = $AuthToken} `
+        -Method Post -ContentType "application/json" -Body (ConvertTo-Json -Depth $CONVERT_TO_JSON_MAX_DEPTH $Request)
+
+    return $Response.'virtual-DNS'.'uuid'
+}
+
+function Remove-ContrailDNSServer {
+    Param (
+        [Parameter(Mandatory = $true)] [string] $ContrailUrl,
+        [Parameter(Mandatory = $true)] [string] $AuthToken,
+        [Parameter(Mandatory = $true)] [string] $DNSServerUuid,
+        [Switch] $Force
+    )
+
+    $DNSServerURl = $ContrailUrl + "/virtual-DNS/" + $DNSServerUuid
+    $DNSServer = Invoke-RestMethod -Method Get -Uri $DNSServerURl -Headers @{"X-Auth-Token" = $AuthToken}
+
+    if ($Force) {
+        $VirtualDNSRecords = $DNSServer.'virtual-DNS'.'virtual_DNS_records'
+        ForEach ($VirtualDNSRecord in $VirtualDNSRecords) {
+            Invoke-RestMethod -Method Delete -Uri $VirtualDNSRecord.'href' `
+                -Headers @{"X-Auth-Token" = $AuthToken} | Out-Null
+        }
+
+        $NetworkIPAMs = $DNSServer.'virtual-DNS'.'network_ipam_back_refs'
+        ForEach ($NetworkIPAM in $NetworkIPAMs) {
+            Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+                -IpamFQName $NetworkIPAM.'to' `
+                -DNSMode 'none'
+        }
+    }
+
+    Invoke-RestMethod -Method Delete -Uri $DNSServerURl -Headers @{"X-Auth-Token" = $AuthToken} | Out-Null
+}
```

Comment:
BTW, have you tried generating test coverage report for this file? If you add CI tag to the Describe block, you can call `Invoke-Selfcheck` with `-CodeCoverage` flag - then you can grep for your files and check what is missing and whether you want to do something about it. 

BTW2 just learned that VScode can't do it in the editor window: https://github.com/PowerShell/vscode-powershell/issues/495 :(

Reply:
Nearly 100%, two constructors are not covered, but they just invoke init method.